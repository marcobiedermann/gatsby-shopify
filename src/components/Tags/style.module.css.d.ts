declare const styles: {
  readonly tags: string;
};
export = styles;
