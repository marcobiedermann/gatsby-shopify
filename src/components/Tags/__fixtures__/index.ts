import { TagsProps } from '..';

const tags: TagsProps = {
  tags: ['tag'],
};

export default tags;
