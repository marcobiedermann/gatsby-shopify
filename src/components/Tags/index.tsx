import React from 'react';
import Tag from '../Tag';
import { tags as tagsStyles } from './style.module.css';

export interface TagsProps {
  tags: string[];
}

function Tags(props: TagsProps) {
  const { tags } = props;

  return (
    <ul className={tagsStyles}>
      {tags.map((tag) => (
        <li key={tag}>
          <Tag>{tag}</Tag>
        </li>
      ))}
    </ul>
  );
}

export default Tags;
