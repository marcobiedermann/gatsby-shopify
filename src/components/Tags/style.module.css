.tags {
  list-style: none;
  margin-bottom: 1.5em;
  margin-top: 0.5em;
  padding: 0;
}
