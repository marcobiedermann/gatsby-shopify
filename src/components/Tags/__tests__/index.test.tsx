import { render } from '@testing-library/react';
import React from 'react';
import Tags from '..';
import tags from '../__fixtures__';

describe('tags component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Tags {...tags} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
