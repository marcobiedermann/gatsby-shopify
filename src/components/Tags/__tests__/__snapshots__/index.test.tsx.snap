// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tags component should render correctly 1`] = `
<ul
  class="tags"
>
  <li>
    <span>
      tag
    </span>
  </li>
</ul>
`;
